### ISSUE TITLE
Check convergence criteria for multinomial regression

### ISSUE THREAD

**Issue by daniel-acuna** [pavanramkumar/pyglmnet#31]
Code below compares scikit learn performance vs pyglmnet for multinomial:

```python
from pyglmnet import *
from sklearn.datasets import make_classification
from sklearn.preprocessing import OneHotEncoder
from sklearn.linear_model import LogisticRegression

X, y = make_classification(n_samples=1000, n_classes=3,
                           n_informative=5, n_features=5,
                           n_redundant=0, random_state=0)
model_mn = GLM(distr='multinomial', alpha=0.01,
               reg_lambda=np.array([0.0]), verbose=False)
model_mn.fit(X, y)
y_pred = model_mn.predict(X, model_mn.fit_params[-1])

lg = LogisticRegression(multi_class='multinomial', solver='newton-cg')
lg.fit(X, y)

print 'sklearn accuracy: {}'.format((lg.predict(X) == y).mean())

y_pred = model_mn.predict(X, model_mn.fit_params[-1]).argmax(1)
print 'pyglmnet accuracy: {}'.format((y_pred == y).mean())

y_ohe = OneHotEncoder(sparse=False).fit_transform(y.reshape(-1, 1))
print 'sklearn -ll: {}'.format(-np.sum(y_ohe*lg.predict_proba(X)))

y_proba = model_mn.predict(X, model_mn.fit_params[-1])
print 'pyglmnet -ll: {}'.format(-np.sum(y_ohe*y_proba))
```

This is the output I am getting right now:

```
sklearn accuracy: 0.806
pyglmnet accuracy: 0.758
sklearn -ll: -698.186411871
pyglmnet -ll: -587.151946163
```

**Comment by hugoguh** [comment]
I tried to run it several times, once it gave exactly the same `0.806` but sometimes it goes under `0.6` even. Probably convergente issues?

**Comment by hugoguh** [comment]
![screen shot 2016-05-02 at 1 46 49 pm](https://cloud.githubusercontent.com/assets/3664942/14964083/5ceb243e-106c-11e6-8813-7b5475e4121b.png)

**Comment by daniel-acuna** [comment]
`pyglmnet` GLM has random start inside that why the numbers are different I think. This issue is more of a mental note to myself because I will fix it later :)

**Comment by hugoguh** [comment]
cool :), 
but still the performance shouldn’t vary so much with the start (from 0.4 to 0.8 accuracy!), convergence needs to be improved (not only for multinomial). Maybe start a new issue for that?

**Comment by daniel-acuna** [comment]
is this happening with other models?

**Comment by hugoguh** [comment]
haven’t checked but if it is a problem with convergence algorithm it might....

**Comment by hugoguh** [comment]
I’ll check for the others

**Comment by jasmainak** [comment]
If you set the `max_iter` to a low number and set the same number in `pyglmnet`, do you get the same result? That way, you are sure it's a convergence issue or not.
